### PR TITLE
Financing Page

### LINKED ISSUE #128: [Financial Page] - Build out financial page
[Financing Page]
* move mortgage calculator to top right of page
* delete yearly taxes and yearly insurance from form
* use current rates found in spreadsheet placeholders

https://www.hubblehomes.com/financing

### PR DESCRIPTION
**Pull request by varunmitra**
Fix #128 

Test URLs:
- Before: https://main--hubblehomes-com--aemsites.hlx.live/financing
- After: https://issue-128--hubblehomes-com--aemsites.hlx.page/financing

### CODE REVIEW

File: blocks/calculator/calculator.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  if (!rate) {
+    document.getElementById('interest_rate').reportValidity();
+    return;
+  }
+  if (!downPayment) {
+    document.getElementById('down_payment').reportValidity();
+    return;
+  }
+  if (!term) {
+    document.getElementById('number_of_years').reportValidity();
```
Comment (samuelopra) [inline blocks/calculator/calculator.js:23]: I do feel like this section can be simplified into something like:

```
  const fields = ['purchase_price', 'interest_rate', 'down_payment', 'number_of_years'];
  const values = {};

  for (const field of fields) {
    const element = document.getElementById(field);
    values[field] = element.value;

    if (!values[field]) {
      element.reportValidity();
      return;
    }
  }
  ```

File: blocks/calculator/calculator.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      textbox.reportValidity();
+    });
+
+    textbox.required = true;
+    formGroup.appendChild(textbox);
+  });
+
+  const calculateButton = document.createElement('button');
+  calculateButton.textContent = 'Calculate Payment';
+  calculateButton.addEventListener('click', calculatePayment);
```
Comment (samuelopra) [inline blocks/calculator/calculator.js:113]: You actually have the fields already, so you can just pass it in here calculatePayment(fields)

File: blocks/calculator/calculator.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+  Math.round(monthlyPayment.toFixed(2));
+
+  const resultContainer = document.getElementById('result');
+  resultContainer.style.display = 'block';
+  resultContainer.innerHTML = `<p>Estimated Monthly Payment = ${formatPrice(monthlyPayment, 'full')}</p>`;
+}
+
+export default async function decorate(block) {
+  const ratesData = await getRatesSheet(null);
```
Comment (bhellema) [inline blocks/calculator/calculator.js:44]: getRateSheet('data') would return you the data node.

File: blocks/calculator/calculator.js (hunk 2 above)
Comment (varunmitra) [reply blocks/calculator/calculator.js:107]: event handlers can't handle parameters. would have to resort to an unnamed function and then call calculate payment.

**Comment by bhellema** [comment]
Down payment shouldn't be larger than purchase price.

image

**Comment by bhellema** [comment]
Clicking calculate should remove the previous payment and only render it on a successful calculation.  It was left over from a previous calculation in this screenshot.

image

**Comment by bhellema** [comment]
Somehow I managed to get a NaN to render 
image

**Comment by bhellema** [comment]
I wonder if we should put limits on things like interest rate and number of years to be < 100.

**Comment by codevoyage** [comment]
The form field's focus colour is different from the original.

**Comment by codevoyage** [comment]
I think the form input on-focus is missing this:
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, .075);

File: blocks/calculator/calculator.css [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+button:hover {
+  background-color: #45a049;
+}
+
+.button-container {
+  width: 20%;
+}
+
+.button-container>button {  
+  color: #003264;
```
Comment (codevoyage) [inline blocks/calculator/calculator.css:59]: There are css variables defined for these colors that can be leveraged.

**Comment by bhellema** [comment]
Can you add `flex-basis: 52px;` to the image-gallery.css file https://github.com/aemsites/hubblehomes-com/blob/main/blocks/image-gallery/image-gallery.css#L91


```
.overlay {
    & .overlay-content {
        & .nav-buttons {
            button {
                width: 52px;
                height: var(--button-height);
                flex-basis: 52px;
            }
        }
    }
}
```

The change to the button style to include flex: 0 has distorted the gallery buttons.  Putting a flex-basis on it should fix it.

**Review by bhellema — APPROVED** [review]

**Review by samuelopra — APPROVED** [review]
👍

**Pull request merged**